src/charm.py

```diff
@@ -485,7 +485,7 @@ def _on_upgrade_charm(self, _event: UpgradeCharmEvent) -> None:
         if not runner_manager:
             return
 
-        runner_manager.flush()
+        runner_manager.flush(wait_repo_check=True)
```

Comment:
Reading the code here without digging into the flush docstring, I would not expect flush to flush busy runners. Could we perhaps just do inside flush what the name of the kwarg suggests, i.e. just wait, and only flush if the `force_flush_busy` kwarg is additionally passed? This might be redundant, but would improve readability.

Reply:
I have changed to use an enum as input. The enum covers different types of strategies.